gcloud/storage/batch.py

```diff
@@ -0,0 +1,208 @@
+# Copyright 2014 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Batch updates / deletes of storage buckets / blobs.
+
+See: https://cloud.google.com/storage/docs/json_api/v1/how-tos/batch
+"""
+from email.encoders import encode_noop
+from email.generator import Generator
+from email.mime.application import MIMEApplication
+from email.mime.multipart import MIMEMultipart
+from email.parser import Parser
+import io
+import json
+import sys
+
+import six
+
+from gcloud._localstack import _LocalStack
+
+
+_BATCHES = _LocalStack()
+
+_PROXIED_ATTRS = [
```

Comment:
Yeah it's certainly the "best" way to proxy and we should stay away from `__getattr***__`.

It just seems like there should be a better way (a la `datastore.Batch`).

The I don't think mapping the `mutation` concept to `storage.batch` is a "best practice".

I'd be much more comfortable with 

- Making protected methods on `storage.Connection` which form, but don't send each request
- Defining `Batch` here as a subclass and using those protected methods to build up `self._requests`
- Not having to patch `connection.http` as a result of dedicated methods

It seems doing this we'll only need to implement

```python
[
    'get_all_buckets',
    'get_bucket',
    'create_bucket',
    'delete_bucket',
]
```